Purchases/Public/RCPurchases.m

```diff
@@ -863,25 +868,30 @@ - (void)cachePurchaserInfo:(RCPurchaserInfo *)info forAppUserID:(NSString *)appU
 }
 
 - (void)updateAllCachesWithCompletionBlock:(nullable RCReceivePurchaserInfoBlock)completion {
-    [self fetchAndCachePurchaserInfoWithCompletion:completion];
-    [self updateOfferingsCache:nil];
+    [self.systemInfo isApplicationBackgroundedWithCompletion:^(BOOL isAppBackgrounded) {
+        [self fetchAndCachePurchaserInfoWithCompletion:completion isAppBackgrounded:isAppBackgrounded];
+        [self updateOfferingsCache:nil isAppBackgrounded:isAppBackgrounded];
+    }];
 }
 
-- (void)fetchAndCachePurchaserInfoWithCompletion:(nullable RCReceivePurchaserInfoBlock)completion {
-    [self.deviceCache setPurchaserInfoCacheTimestampToNow];
+- (void)fetchAndCachePurchaserInfoWithCompletion:(nullable RCReceivePurchaserInfoBlock)completion
+                               isAppBackgrounded:(BOOL)isAppBackgrounded {
     NSString *appUserID = self.identityManager.currentAppUserID;
-    [self.backend getSubscriberDataWithAppUserID:appUserID
-                                      completion:^(RCPurchaserInfo * _Nullable info,
-                                                   NSError * _Nullable error) {
-                                          if (error == nil) {
-                                              [self cachePurchaserInfo:info forAppUserID:appUserID];
-                                              [self sendUpdatedPurchaserInfoToDelegateIfChanged:info];
-                                          } else {
-                                              [self.deviceCache clearPurchaserInfoCacheTimestamp];
-                                          }
-
-                                          CALL_IF_SET_ON_MAIN_THREAD(completion, info, error);
-                                      }];
+    [self.deviceCache setPurchaserInfoCacheTimestampToNowForAppUserID:appUserID];
+    [self.operationDispatcher dispatchOnWorkerThreadWithRandomDelay:isAppBackgrounded :^{
```

Comment:
this API came out a bit weird - note the nameless second param that just starts with `:`